core/frontend/src/tile/TileTreeReference.ts

```diff
@@ -288,6 +293,14 @@ export abstract class TileTreeReference /* implements RenderMemory.Consumer */ {
    */
   protected _collectTileGeometry(collector: TileGeometryCollector): void {
     const tree = this.treeOwner.load();
+
+    // The reprojectGeometry property is only relevant for RealityTileTree because only reality tiles have the reprojectionTransform.
+    if (tree instanceof RealityTileTree) {
+      // Make sure tree knows to reproject when collectTileGeometry is called.
+      // this.reprojectGeometry will only be defined if this is a GeometryTileTreeReference.
+      tree.reprojectGeometry = this.reprojectGeometry;
```

Comment:
Eww. There's no alternative to (1) up-casting and (2) modifying the `TileTree`'s state here? I think my suggestion of putting the "reproject" option into the tile tree Id would solve both of these (`tree.reprojectGeometry` would be set in the tree's constructor).